pulsar-broker/src/test/java/com/yahoo/pulsar/client/impl/BrokerClientIntegrationTest.java

```diff
@@ -517,6 +523,65 @@ public void testCloseConnectionOnBrokerRejectedRequest() throws Exception {
     }
 
     /**
+     * It verifies that broker throttles down configured concurrent topic loading requests
+     * 
+     * <pre>
+     * 1. Start broker with N maxConcurrentTopicLoadRequest 
+     * 2. create concurrent producers on different topics which makes broker to load topics concurrently
+     * 3. Producer operationtimeout = 1 ms so, if producers creation will fail for throttled topics
+     * 4. verify all producers should have connected
+     * </pre>
+     * 
+     * @throws Exception
+     */
+    @Test(timeOut = 5000)
+    public void testMaxConcurrentTopicLoading() throws Exception {
+
+        final PulsarClient pulsarClient;
+        final PulsarClient pulsarClient2;
+
+        final String topicName = "persistent://prop/usw/my-ns/newTopic";
+
+        final int concurrentLookupRequests = 20;
+        ClientConfiguration clientConf = new ClientConfiguration();
+        clientConf.setMaxNumberOfRejectedRequestPerConnection(0);
+        clientConf.setOperationTimeout(1, TimeUnit.MILLISECONDS);
+        clientConf.setStatsInterval(0, TimeUnit.SECONDS);
+        stopBroker();
+        pulsar.getConfiguration().setMaxConcurrentTopicLoadRequest(1);
+        startBroker();
+        String lookupUrl = new URI("pulsar://localhost:" + BROKER_PORT).toString();
+        pulsarClient = PulsarClient.create(lookupUrl, clientConf);
+
+        ClientConfiguration clientConf2 = new ClientConfiguration();
+        clientConf2.setStatsInterval(0, TimeUnit.SECONDS);
+        clientConf2.setIoThreads(concurrentLookupRequests);
+        clientConf2.setConnectionsPerBroker(20);
+        clientConf2.setOperationTimeout(1, TimeUnit.MILLISECONDS);
+        pulsarClient2 = PulsarClient.create(lookupUrl, clientConf2);
+
+        ProducerImpl producer = (ProducerImpl) pulsarClient.createProducer(topicName);
+        ClientCnx cnx = producer.cnx();
+        assertTrue(cnx.channel().isActive());
+        ExecutorService executor = Executors.newFixedThreadPool(concurrentLookupRequests);
+        List<CompletableFuture<Producer>> futures = Lists.newArrayList();
+        final int totalProducers = 10;
+        CountDownLatch latch = new CountDownLatch(totalProducers);
+        for (int i = 0; i < totalProducers; i++) {
+            executor.submit(() -> {
+                futures.add(pulsarClient2.createProducerAsync(topicName + randomUUID().toString()));
+                futures.add(pulsarClient.createProducerAsync(topicName + randomUUID().toString()));
+                latch.countDown();
+            });
+        }
+
+        latch.await();
+        FutureUtil.waitForAll(futures).get();
+
+        pulsarClient.close();
```

Comment:
actually as we decided broker should not give `TooManyReqeustsException` to client and put the topic in to the queue so, it give chance to unused topic to be loaded for stats visibility. So, eventually all the producers should get connected even it gets error internally at broker.